File: src/types/audit.ts
export interface AuditEntry {
  id: string;
  entity_type: string;
  entity_id: string;
  action: string;
  summary: string;
  details: string;
  created_at: string;
}

export interface AuditFilters {
  entity_type?: string;
  entity_id?: string;
  action?: string;
  limit?: number;
}

export interface NotificationSummary {
  active_incidents: number;
  overdue_action_items: number;
  sla_breaches: number;
  recent_audit_count: number;
}

File: src-tauri/src/models/audit.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AuditEntry {
    pub id: String,
    pub entity_type: String,
    pub entity_id: String,
    pub action: String,
    pub summary: String,
    pub details: String,
    pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, Default)]
pub struct AuditFilters {
    pub entity_type: Option<String>,
    pub entity_id: Option<String>,
    pub action: Option<String>,
    pub limit: Option<i64>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct NotificationSummary {
    pub active_incidents: i64,
    pub overdue_action_items: i64,
    pub sla_breaches: i64,
    pub recent_audit_count: i64,
}
